### ISSUE TITLE
`null` is not working properly on nested keys

### ISSUE THREAD

**Issue by jayrepo** [taverntesting/tavern#56]
```yml
test_name: Get some fake data from the JSON placeholder API

stages:
  - name: Verify only id
    request:
      url: https://reqres.in/api/users/1
      method: GET
    response:
      status_code: 200
      body:
        data:
          id: 1
          first_name: null
          last_name: null
          avatar: null
```
The above test failed with error:
 Value mismatch: got '{'id': 1, 'first_name': 'George', 'last_name': 'Bluth', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'}' (type: <class 'dict'>), expected '{'id': 1, 'first_name': None,
'last_name': None, 'avatar': None}' (type: <class 'dict'>)

**Comment by michaelboulton** [comment]
This should be fixed in version 0.7.0. From this version onwards we are going to suggest using a magic `!anything` in the response to match any values. Full documentation is here: https://taverntesting.github.io/documentation#matching-arbitrary-return-values-in-a-response

I considered adding something like `!anylist`/`!anymapping` as well to match and list or mapping, but I think the suggestion for the time being is that if you need to say "I expect a list of any values to be returned", you should use the pykwalify extension function.

**Comment by jayrepo** [comment]
`!anything` is not working even on `data: !anything` level
please see log below:
```log
2018-03-09 23:06:24,570 [INFO]: (tavern.response.rest:67) Response: '<Response [200]>' ({"data":{"id":1,"first_name":"George","last_name":"Bluth","avatar":"https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg"}})
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:192) Nothing expected to save for body
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:192) Nothing expected to save for headers
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:192) Nothing expected to save for redirect_query_params
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:122) No save function
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:170) Validating body for {'data': <object object at 0x000001AEA6C99400>}
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.base:122) expected = {'data': <object object at 0x000001AEA6C99400>}, actual = {'data': {'id': 1, 'first_name': 'George', 'last_name': 'Bluth', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'}}
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.base:131) data: <object object at 0x000001AEA6C99400> vs {'id': 1, 'first_name': 'George', 'last_name': 'Bluth', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'}
2018-03-09 23:06:24,570 [ERROR]: (tavern.response.base:136) Key mismatch on data
2018-03-09 23:06:24,570 [ERROR]: (tavern.response.base:37) Value mismatch: got '{'id': 1, 'first_name': 'George', 'last_name': 'Bluth', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'}' (type: <class 'dict'>), expected '<object object at 0x000001AEA6C99400>' (type: <class 'object'>)
Traceback (most recent call last):
  File "c:\python36\lib\site-packages\tavern\response\base.py", line 134, in recurse_check_key_match
    self._check_response_keys_recursive(expected_val, actual_val, [])
  File "c:\python36\lib\site-packages\tavern\response\base.py", line 80, in _check_response_keys_recursive
    raise exceptions.KeyMismatchError("Structure of returned data was different than expected ({})".format(full_err()))
tavern.util.exceptions.KeyMismatchError: Structure of returned data was different than expected (expected = '<object object at 0x000001AEA6C99400>', actual = '{'id': 1, 'first_name': 'George', 'last_name': 'Bluth', 'avatar': 'https://s3.amazonaws.com/uifaces/faces/twitter/calebogden/128.jpg'}')
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:170) Validating headers for {}
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.base:112) Nothing to check against
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.rest:170) Validating redirect_query_params for {}
2018-03-09 23:06:24,570 [DEBUG]: (tavern.response.base:112) Nothing to check against
2018-03-09 23:06:24,586 [ERROR]: (tavern.printer:21) FAILED: Verify first name not include number [200]
2018-03-09 23:06:24,586 [ERROR]: (tavern.printer:22) Expected: {'requests': {'status_code': 200, 'body': {'data': <object object at 0x000001AEA6C99400>}}}
```

**Comment by michaelboulton** [comment]
Sorry for the late reply, I realised the documentation is a big ambiguous. At the moment `!anything` will only match 1 value, not a dictionary/list.

This might be changed in the future so that `!anything` truly matches anything, and that `!int`/`!str` match an integer/string

**Comment by michaelboulton** [comment]
This should be fixed in 0.8.0, There are also some new magic variables for matching specific types like `!anyint` and `!anyfloat` but I don't have time to update the documentation at the moment

**Comment by James-KJiang** [comment]
Hi michaelboulton, I am a Chinese tester and very glad to find this useful tool to do the API test. For this "null" value, how I can verify a field is not returned from server.
for example:

`test_name: Get some fake data from the JSON placeholder API

stages:
  - name: Verify only id
    request:
      url: https://reqres.in/api/users/1
      method: GET
    response:
      status_code: 200
      body:
        data:
          id: 1
          first_name: null
          last_name: null
          avatar: null`
How can I make sure the 'first_name' is not return in the response from server? "null" value is not work, what should I do? I really hope to get your help!
